patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.TML.cs

```diff
@@ -314,8 +321,12 @@ internal List<ModDownloadItem> QueryItemsSynchronously(out List<string> missingM
 
 						for (int j = 0; j < _queryReturnCount; j++) {
 							var itemsIndex = j + i * Constants.kNumUGCResultsPerPage;
-							var item = GenerateModDownloadItemFromQuery((uint)j);
-							if (item is null) {
+							var match = TryGenerateModDownloadItem((uint)j, out var item);
+							if (match != WorkshopSearchReturnState.Success) {
+								if (match == WorkshopSearchReturnState.Banned) {
+									throw new BannedModException($"The mod {item.DisplayName} with ID {item.PublishId} is Banned!", item.DisplayName, item.PublishId.ToString());
```

Comment:
The `message` here should probably be generated inside the `BannedModException` class, and should probably be localized. Also no need for braces for single line guard `if`